pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -57,7 +57,7 @@ type mutableListenerOpts struct {
 
 func (configgen *ConfigGeneratorImpl) buildGatewayListeners(builder *ListenerBuilder) *ListenerBuilder {
 	if builder.node.MergedGateway == nil {
-		log.Debugf("buildGatewayListeners: no gateways for router %v", builder.node.ID)
```

Comment:
It is deliberate not to print spammy logs

Reply:
When gateway resources don't match the target proxy,  the proxy will get no lds in the startup or lose all lds in runtime. So i think these aren't spammy logs. Maybe need to warn users and report what happened.

Reply:
add a new metric for this?

Reply:
+1 to Zhonghu, its not an appropriate way to expose this info